### PR TITLE
fix: flush headers if express handles request

### LINKED ISSUE #13: Headers set by fastify are ignored if `express` handles request
## 🐛 Bug Report

If I set headers via some hook in fastify, they are not included in the response if an express handler handles the request.

This is not true the other way around - if an express middleware sets a header it's included in the response no matter of it's express or fastify the responds.

## To Reproduce

```js
const fastify = require('fastify')();
const fastifyExpress = require('fastify-express');
const fp = require('fastify-plugin');
const express = require('express');

fastify.register(
  fp(
    (instance, _, next) => {
      instance.addHook('preValidation', (_, reply, done) => {
        reply.headers({
          foo: 'bar',
        });

        done();
      });

      next();
    },
    { fastify: '3.x' },
  ),
);

fastify.register(fastifyExpress).after(() => {
  const router = express.Router();

  router.use('/express', (_, res) => {
    res.sendStatus(200);
  });

  fastify.use(router);
});

fastify.get('/fastify', (_, reply) => {
  reply.code(200).send();
});

fastify.listen(3000);
```

```sh-session
$ curl -i http://localhost:3000/fastify 
HTTP/1.1 200 OK
foo: bar
content-length: 0
Date: Thu, 22 Oct 2020 10:48:01 GMT
Connection: keep-alive
Keep-Alive: timeout=5

$ curl -i http://localhost:3000/express
HTTP/1.1 200 OK
Content-Type: text/plain; charset=utf-8
Content-Length: 2
ETag: W/"2-nOO9QiTIwXgNtWtBJezz8kv3SLc"
Date: Thu, 22 Oct 2020 10:48:03 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

FWIW, using `reply.raw.setHeader` doesn't work either.

## Expected behavior

I would expect the header to be written to the response regardless of express handling the request or not.

This makes it impossible to migrate parts of our app piecemeal to Fastify as the headers set (specifically HSTS headers now) are ignored.

## Your Environment

- *node version*: v12.19.0
- *fastify version*: 3.7.0
- *os*: Mac
- *any other relevant information*

### PR DESCRIPTION
**Pull request by SimenB**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Fixes #13

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ function expressPlugin (fastify, options, next) {
   }
 
   function runConnect (req, reply, next) {
+    for (const [headerName, headerValue] of Object.entries(reply.getHeaders())) {
```
Comment (SimenB) [inline index.js:55]: should it be moved inside the `if`?

Comment (mcollina) [reply index.js:55]: Yes it should.

Comment (SimenB) [reply index.js]: done 👍

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**